.github/workflows/main.yml

```diff
@@ -0,0 +1,59 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  java:
+    name: Java/Gradle
+    runs-on: ubuntu-22.04
+    strategy:
+      max-parallel: 4
+      matrix:
+        java-version: [11, 17, 18]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+
+      - name: Set up JDK ${{ matrix.java-version }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java-version }}
+
+      - name: Spotless Check
+        uses: gradle/gradle-build-action@v2
+        with:
+          # Spotless must run in a different invocation, because
+          # it has some weird Gradle configuration/variant issue
+          arguments: spotlessCheck --scan
+
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: --rerun-tasks assemble ${{ env.ADDITIONAL_GRADLE_OPTS }} check publishToMavenLocal -x jmh -x spotlessCheck --scan
```

Comment:
still uses `jmh` wouldnt it fail if that task does not exist?
did you try running these commands locally?

because it seems like CI doesnt run on this PR as is (maybe the workflow needs to be merged first)

but this might mean we need to iterate a lot until we have ironed out all the bugs

Reply:
True. The spotless also not present. It will be inherited by adding a `buildSrc` module + gradle version catalog. 

As of now. I think only `gradle --rerun-tasks assemble check publishToMavenLocal --scan` passes locally.